utils/index.js

```diff
@@ -38,31 +44,76 @@ const filterObject = function (object, keys) {
 // passed in or create a sub-object on the binding and
 // then call setKey recursively to set the value
 const setKey = function (binding, key, value) {
-  if (!key) return;
+  const type = getType(key);
 
-  if (isString(key)) {
+  const isString = function () {
+    if (!key) return;
     binding[key] = value;
-    return;
-  }
-  if (isArray(key)) {
+  };
+
+  const isArray = function () {
     binding[key[0]] = new Array(value);
-    return;
-  }
-  if (isObject(key)) {
+  };
+
+  const isObject = function () {
     for (const subkey in key) {
       if (!binding[subkey]) {
         binding[subkey] = {};
       }
       setKey(binding[subkey], key[subkey], value);
     }
-  }
+  };
+
+  const bindings = {
+    String: isString,
+    Array: isArray,
+    Object: isObject,
+    default: function () {}
+  };
+  return (bindings[type] || bindings.default)();
 };
 
+function getKeyMapping({ client, clientInfo, filename }) {
+  if (!client) {
+    return filename;
+  }
+  if (clientInfo.mapping[filename] || clientInfo.mapping[filename] === '') {
+    return clientInfo.mapping[filename].key || clientInfo.mapping[filename];
+  }
+  return filename;
+}
+
+function getValueMapping({ client, clientInfo, bindingsRoot, filename, key }) {
+  const filepath = path.join(bindingsRoot, filename);
+
+  const fileContent = fs.readFileSync(filepath).toString().trim();
+
+  if (!client) {
+    return fileContent;
+  }
+
+  if (clientInfo.mapping[filename] && clientInfo.mapping[filename].path) {
+    if (clientInfo.mapping[filename].copy) {
+      const newFilepath = path.join('.', path.basename(filepath));
```

Comment:
@mhdawson I updated it to use something like tmp-fdDkcel/tls.key where "fdDkcel" is a random string. Is that what you mean?

Reply:
compared to previous implementation, this one will not override in case of binding two times, instead it will create another tmp-.... file.

Reply:
Another solution could be having something like tmp-uuid where uuid is going to be the same each time, in order to override directory instead of creating a new one. Purpose of uuid, being clearly a unique directroy that wont conflict with other directories.